src/Engine/GameVersion.java

```diff
@@ -0,0 +1,60 @@
+package Engine;
+
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.io.Serializable;
+
+public class GameVersion implements Serializable
+{
+  private static final long serialVersionUID = 1L; // This class should always keep the same serialization format.
+  
+  int majorRev, minorRev, hotfix;
+  public GameVersion()
+  {
+    majorRev = 0;
+    minorRev = 1;
+    hotfix = 0;
```

Comment:
I feel like we can probably claim credit for something greater than `0:1:0`. What feature set do we want to support to declare 1.0?

That also brings up the more general question of what constitutes a major/minor/hotfix change?
If `majorRev` is supposed to be like the differences between game sequels, then we are (probably) never gonna roll it past 1, and I wonder if it's worth recording.

Reply:
Hmm.
I suppose this also opens the possible can of worms about save compatibility; I think we can probably default everything's version ID and just let GameVersion take over logic to decide whether stuff's compatible.
Under that scheme, I'd propose `majorRev` be "substantial/breaking changes to game logic", while `minorRev` would be non-breaking/UI changes and `hotfix` would be quick bug fixes (they've happened a few times, though obviously I try to make sure they never happen).

For instance, adding a new CO that uses only existing features would be a `minorRev` change since the engine will be backwards-compatible, though saves with the new CO obviously wouldn't be.

Reply:
Forgot to note: if we go with my above scheme, I'd call it version 3.0.0, probably.

Reply:
That all sounds pretty reasonable. Not sure how you get 3:0:0, but fine with me.
So would we like, prompt them if the versions differ, and let them try loading it anyway? Or just always try and notify on failure?

Reply:
My thinking was that we'd auto fail on `majorRev` difference, but try if there's a `minorRev` difference.
I was gonna say only try to load older saves, but I guess there's no real reason to require that other than "new content that just won't load anyway".

Reply:
Oh, and I got a `majorRev` of 3 from Jay's previously-imposed versioning. I could like, go back and count, but that sounds like work for no benefit.

Reply:
Nah, mostly curious. I think he called the last one 1.2, and was gonna advance to 1.3? Whatever, I'm good with 3:0:0. Should we display the version number somewhere (maybe in the corner of the first menu, a la Dawn of War)? At least log it to the console on startup, I think.

Reply:
Yeah, he was. Naming it `1.3` goes back to your point of "is it really worth recording a number that will never increment?", though.

My intent was to put it on the title bar, but I forgot. Or maybe I was lazy.

...wait a second. Versioning brings up *another* weirdness. What of the test branch? My assumption was that it was the only one that would be getting "releases", but `master` will have the same versions and that... doesn't seem right, as they're incompatible due to a number of mechanical differences and a lot of added content.

Reply:
It seems like the test branch will probably increment versions at about the same rate as master. We could just let them align, though that means master would choke on saves from the test branch.

Could also just rev the minor version for the test branch. That way they are always different, but it's clear they are related.

It might also be good to add a notional idea of the roll criteria for each number as a comment in the file.

Reply:
Another possibility: `majorRev` is game branch (so `master` or `test`), `minorRev` would be breaking changes, `hotfix` non-breaking.
Could be argued any way, but I suppose it prooobably doesn't matter all that much right now since the only one being given to users is the test branch, at this point.